### PR TITLE
FAQ entry on ambiguous mappings

### PR DESCRIPTION
**Pull request by sjaakd**
Hey Filip, is it possible to make "local"  references so that the URL's jump to the right spot:

e.g.:

https://mapstruct.org/faq/#ambiguous

### CODE REVIEW

File: content/faq/_index.md [hunk 1]
```diff
@@ -266,6 +266,26 @@ More detailed information can be found in the reference guide.
 
 {{% /faq_question %}}
 
+{{% faq_question "How to solve ambiguous methods" %}}
+
+MapStruct tries various mechanisms to map a sourceproperty to a taregtproperty when it cannot make a direct mapping. In order, MapStruct tries:
+
+1. other mapping methods (inside the mapper, or via the uses relation). This concerns both hand-written methods and MapStruct generated methods
+2. direct mapping (soure type is directly assignable to target type)
```
Comment (filiphr) [inline content/faq/_index.md:274]:
```suggestion
2. direct mapping (source type is directly assignable to target type)
```

File: content/faq/_index.md [hunk 2]
```diff
@@ -266,6 +266,26 @@ More detailed information can be found in the reference guide.
...
+3. type conversions (e.g. from int to String)
+4. BuiltIn methods (see documentation on a list of supported methods)
+5. 2 step method, variant 1: `target = methodY( methodX ( source ) )`
+6. 2 step method, variant 2: `target = methodY( conversionX ( source ) )`
+7. 2 step method, variant 3: `target = conversionY( methodX ( source ) )`
+
+Whenever MapStruct finds a unique candidate, MapStruct stops and uses this method to make the mapping between source and target. However, for option 1, 5, 6, 7 it is possible that multiple egible candidtates are found for which MapStruct cannot decide to make select one. MapStruct reports this as "ambiguous mapping method" and lists the methods from which it cannot make a selection. Here, you have to guide MapStruct making the correct mapping.
```
Comment (filiphr) [inline content/faq/_index.md:281]:
```suggestion
Whenever MapStruct finds a unique candidate, MapStruct stops and uses this method to make the mapping between source and target. However, for option 1, 5, 6, 7 it is possible that multiple eligible candidates are found for which MapStruct cannot decide which one to select. MapStruct reports this as "ambiguous mapping method" and lists the methods from which it cannot make a selection. Here, you have to guide MapStruct in making the correct mapping.
```

File: content/faq/_index.md [hunk 3]
```diff
@@ -266,6 +266,26 @@ More detailed information can be found in the reference guide.
...
+
+This can be done in the following ways:
+* provide an method with the exact signature if MapStruct cannot select between base- and super types.
```
Comment (filiphr) [inline content/faq/_index.md:284]:
```suggestion
* provide a method with the exact signature if MapStruct cannot select between base- and super types.
```

**Comment by sjaakd** [comment]
Done.. Thanks for the feedback

**Pull request merged**